### ISSUE TITLE
Tokenize clients

### ISSUE THREAD

**Issue by dalebezolli** [dalebezolli/cowatch#8]
#### Description
We need to create a public & private token pair for each client.
The public token 
- allows for easier handling of actions from one client to another.

The private token 
- allows for privileged requests by a user
- eases management of resources of a client even if there were accidental disconnections

#### Additional Information
For simplicity the public & private token should be created when the user first opens the extension & he doesn't have a token yet.
Ideally we should at some point link these tokens with google accounts and collect user information that way, but for now the existing method will suffice.

Right after the initial connection, the client must send an authorization request. As we don't hold user information this will only take the username and image of the user initially and will automatically generate a public and private token.

If any other request is run without a private token attached to it, the request will fail regardless of if we have a valid connection.

#### Task List
To recap we need the following changes:
- [x] Creation of authorization request in the server (use `UpdateClientDetails()` to update the initial details of a user)
- [x] Remove `UpdateClientDetails()` from other requests
- [x] Update of existing requests to expect authorization
- [x] Update of `updateRoomClientsWithLatestChanges()` to send the public token of all joined users.
- [x] Update of extension to send an authorization request & store the tokens after initial connection
